### PR TITLE
feat:  Add llama.cpp web assembly support

### PR DESCRIPTION
**Pull request by GordonSmith**
Primarily for calculating embeddings on the client

### CODE REVIEW

File: packages/llama/esbuild.mjs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { browserTpl, neutralTpl, nodeTpl } from "../../../Visualization/packages/esbuild-plugins/dist/index.js";
```
Comment (GordonSmith) [inline packages/llama/esbuild.mjs:1]: Need to release latest Viz for this...

File: packages/llama/test/llama.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        expect(v).to.be.not.empty;
+        expect(v).equals(v1);
+        Llama.unload();
+
+        llama = await Llama.load();
+        v = llama.version();
+        expect(v).to.be.a.string;
+        expect(v).to.be.not.empty;
+        expect(v).equals(v1);
+        Llama.unload();
```
Comment (jeclrsg) [inline packages/llama/test/llama.ts:25]: Is this supposed to perform the same test twice?

**Review by jeclrsg — COMMENTED** [review]
@GordonSmith only one question about the test file. The version test seems to perform the exact same thing twice?

File: packages/llama/test/llama.ts (hunk 2 above)
Comment (GordonSmith) [reply packages/llama/test/llama.ts:25]: Yes - its testing the unload and reload capabilities.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**